src/coreComponents/mesh/ElementRegionManager.cpp

```diff
@@ -152,6 +154,78 @@ void ElementRegionManager::generateMesh( CellBlockManagerABC const & cellBlockMa
   } );
 
 }
+void ElementRegionManager::checkSubRegionRegistering( CellBlockManagerABC const & cellBlockManager )
```

Comment:
In `CellElementRegion::computeSelectedCellBlocks`, you already compute the `std::set` of the selected `cellBlocks`. In the same time, could you gather the `std::set` of all the available `cellBlocks` and check that `available == selected`?
For the duplication of selection, instead of storing the selected cell blocks in a flat `std::set`, use a `std::map` that binds the `matcher` to the `std::set` that matches. Then you have a finer grain that you can use to verify that `std::set_intersection` is empty before merging?

Reply:
Do you think that I should store the selected cellblocks in a dedicated `std::set` attribute?

Reply:
I would have to double check, but I'm not sure that you have to store them (as an attribute).
Once you've done your allocation (through the function), keep track of it in the function and perform the check at the exact same moment?

Reply:
So:
- moved the duplication detection as soon as possible, in `CellElementRegion::generateMesh()`,
- to keep track of the other matching cellblock, changed the `CellElementRegion::generateMesh()` signature to compare / modify a `std::map< string, CellElementRegion const * > & cellBlocksMatchers` between calls,
- at `ElementRegionManager`, we now only have the "orphan" check.